buildSrc/src/main/kotlin/KotlinCommunity.kt

```diff
@@ -37,4 +37,19 @@ fun addDevRepositoryIfEnabled(repositoryHandler: RepositoryHandler, project: Pro
     repositoryHandler.maven {
         url = URI.create(devRepoUrl)
     }
+}
+
+/**
+ * Should be used for running against non-released Kotlin compiler on a system test level.
+ *
+ * @return a Kotlin API version parametrized from command line or gradle.properties, null otherwise
+ */
+fun getOverriddenKotlinNativeVersion(project: Project): String? {
+    val nativeVersion = project.rootProject.properties["kotlin.native.version"] as? String
```

Comment:
```suggestion
    val nativeVersion = project.providers.gradleProperty("kotlin.native.version").orNull
```

I suggest using the Provider API instead of properties.